### ISSUE TITLE
[Bug] Destructive Spatial Feature Squashing via AdaptiveMaxPool2d((5, 5))

### ISSUE THREAD

**Issue by RustamSheoran** [Mohammedmt10/self-driving-model-in-simulation#1]
## Labels
`bug`, `model-architecture`, `computer-vision`

## Affected Files
- `model.py` (lines 42–46)

## Problem Summary
The convolutional vision branch reduces a $256 \times 30 \times 60$ feature map down to $5 \times 5$ in a single step using `AdaptiveMaxPool2d((5, 5))`.

## Root Cause Code
```python
# model.py lines 39-44
nn.Conv2d(256, 256, kernel_size=3, stride=1, padding=1),
nn.BatchNorm2d(256),
nn.ReLU(),
nn.MaxPool2d(kernel_size=2, stride=2),   # down to (256, 30, 60)
nn.AdaptiveMaxPool2d((5, 5)),            # squashed directly to (256, 5, 5)
nn.Flatten(),                            # 256 * 5 * 5 = 6400
```

## Why It Is Completely Ruining Everything
1. **Massive Information Loss:** Downsampling from $30 \times 60$ to $5 \times 5$ via max-pooling compresses a $6 \times 12$ feature region into a single maximum value, discarding **98.6% of spatial activations** (71 out of 72 values) in one step.
2. **Aspect Ratio Distortion:** The feature map aspect ratio changes abruptly from 2:1 ($30 \times 60$) to 1:1 ($5 \times 5$), distorting horizontal resolution twice as severely as vertical resolution.
3. This obliterates spatial localization required for detecting lane boundary distances, road curvature, and distant hazards.

## Proposed Fix
Add an additional standard convolutional downsampling block (e.g. stride-2 conv) and use `AdaptiveAvgPool2d((4, 8))` to preserve the 1:2 aspect ratio, or adopt a standard backbone (ResNet-18 / RegNet).
